Convert six API classes to use generated serialization

brentfulgham opened the pull request:
#### 7e720e06b1178d479f844eb87e99db007fd370cf
<pre>
Convert six API classes to use generated serialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=249799">https://bugs.webkit.org/show_bug.cgi?id=249799</a>
&lt;rdar://problem/103646493&gt;

Reviewed by Alex Christensen.

Convert APIError, APIFrameHandle, APIPageHandle, APIURL, APIURLRequest, and APIURLResponse
types to use the new generated serializers.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/API/APIError.cpp:
(API::Error::encode const): Deleted.
(API::Error::decode): Deleted.
* Source/WebKit/Shared/API/APIError.h:
* Source/WebKit/Shared/API/APIError.serialization.in: Added.
* Source/WebKit/Shared/API/APIFrameHandle.cpp: Removed.
* Source/WebKit/Shared/API/APIFrameHandle.h:
* Source/WebKit/Shared/API/APIFrameHandle.serialization.in: Added.
* Source/WebKit/Shared/API/APIPageHandle.cpp: Removed.
* Source/WebKit/Shared/API/APIPageHandle.h:
(API::PageHandle::create):
(API::PageHandle::createAutoconverting):
(API::PageHandle::PageHandle):
* Source/WebKit/Shared/API/APIPageHandle.serialization.in: Added.
* Source/WebKit/Shared/API/APIURL.h:
(API::URL::encode const): Deleted.
(API::URL::decode): Deleted.
* Source/WebKit/Shared/API/APIURL.serialization.in: Added.
* Source/WebKit/Shared/API/APIURLRequest.cpp:
(API::URLRequest::encode const): Deleted.
(API::URLRequest::decode): Deleted.
* Source/WebKit/Shared/API/APIURLRequest.h:
* Source/WebKit/Shared/API/APIURLRequest.serialization.in: Added.
* Source/WebKit/Shared/API/APIURLResponse.cpp: Removed.
* Source/WebKit/Shared/API/APIURLResponse.h:
(API::URLResponse::URLResponse):
* Source/WebKit/Shared/API/APIURLResponse.serialization.in: Added.
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp:
* Source/WebKit/Shared/UserData.cpp:
(WebKit::UserData::encode):
(WebKit::UserData::decode):
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/258400@main">https://commits.webkit.org/258400@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/722e640a012bbcb78c14642017d34ba0963a59cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101844 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10993 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34919 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111181 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11954 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1907 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108939 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107624 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9145 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/92413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91030 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/4585 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/25327 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4667 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1763 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/10750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/44814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5755 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6418 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/DerivedSources.make:
```diff
@@ -473,7 +473,13 @@ SERIALIZATION_DESCRIPTION_FILES = \
 	GPUProcess/media/TextTrackPrivateRemoteConfiguration.serialization.in \
 	NetworkProcess/NetworkProcessCreationParameters.serialization.in \
+	Shared/API/APIError.serialization.in \
+    Shared/API/APIFrameHandle.serialization.in \
```
indentation

brentfulgham replied:
Rats!

achristensen07 commented on Source/WebKit/Shared/API/APIFrameHandle.h:
```diff
@@ -29,26 +29,32 @@
+        return adoptRef(*new FrameHandle(frameID, autoconverting));
+    }
+
+    explicit FrameHandle(WebCore::FrameIdentifier frameID, bool isAutoconverting)
```
This constructor should be private.

brentfulgham replied:
That doesn't work because APIObject's `constructInWrapper` expects access to the constructor. I don't understand the memory handling of that function, so don't easily see how to change it to use a `create` method.

Commit:
Convert six API classes to use generated serialization
https://bugs.webkit.org/show_bug.cgi?id=249799
<rdar://problem/103646493>

Reviewed by Alex Christensen.

Convert APIError, APIFrameHandle, APIPageHandle, APIURL, APIURLRequest, and APIURLResponse
types to use the new generated serializers.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/API/APIError.cpp:
(API::Error::encode const): Deleted.
(API::Error::decode): Deleted.
* Source/WebKit/Shared/API/APIError.h:
* Source/WebKit/Shared/API/APIError.serialization.in: Added.
* Source/WebKit/Shared/API/APIFrameHandle.cpp: Removed.
* Source/WebKit/Shared/API/APIFrameHandle.h:
* Source/WebKit/Shared/API/APIFrameHandle.serialization.in: Added.
* Source/WebKit/Shared/API/APIPageHandle.cpp: Removed.
* Source/WebKit/Shared/API/APIPageHandle.h:
(API::PageHandle::create):
(API::PageHandle::createAutoconverting):
(API::PageHandle::PageHandle):
* Source/WebKit/Shared/API/APIPageHandle.serialization.in: Added.
* Source/WebKit/Shared/API/APIURL.h:
(API::URL::encode const): Deleted.
(API::URL::decode): Deleted.
* Source/WebKit/Shared/API/APIURL.serialization.in: Added.
* Source/WebKit/Shared/API/APIURLRequest.cpp:
(API::URLRequest::encode const): Deleted.
(API::URLRequest::decode): Deleted.
* Source/WebKit/Shared/API/APIURLRequest.h:
* Source/WebKit/Shared/API/APIURLRequest.serialization.in: Added.
* Source/WebKit/Shared/API/APIURLResponse.cpp: Removed.
* Source/WebKit/Shared/API/APIURLResponse.h:
(API::URLResponse::URLResponse):
* Source/WebKit/Shared/API/APIURLResponse.serialization.in: Added.
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp:
* Source/WebKit/Shared/UserData.cpp:
(WebKit::UserData::encode):
(WebKit::UserData::decode):
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/258400@main
